### PR TITLE
style: start styling docs site

### PR DESCRIPTION
**Pull request by macklinu**
Relates to #66 

This branch is a start at styling the MDX doc layout.

image

I grabbed snippets from and learned about building out design system components from the Rebass codebase (especially [`@rebass/markdown`](https://github.com/jxnblk/rebass/blob/master/markdown/index.js)), so it looks a lot like Rebass but isn't technically right now. Open to talking through what the site should look like here or in another issue, but it helps me move forward when I can visualize something and go from there, and also it's a little bit more fun to write docs when the output is a little bit prettier. 😄

### CODE REVIEW

File: examples/docs/src/components.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+/* eslint-disable react/display-name */
+
+import * as React from "react";
+import system from "system-components/emotion";
```
Comment (macklinu) [inline examples/docs/src/components.js]: This uses a lot of [system-components](https://github.com/jxnblk/styled-system/tree/master/system-components), which is shorter way of writing [styled-system](https://github.com/jxnblk/styled-system) components. Thoughts on using this in the docs site?

Comment (ChristopherBiscardi) [reply examples/docs/src/components.js]: Works for me. I'm a huge fan of styled-system+emotion, so happy to use system-components. I don't have anything against using rebass either.

**Comment by ChristopherBiscardi** [comment]
We should move global context providers like the `ThemeProvider` and possibly `MDXProvider` into the new [wrapRootElement](https://github.com/ChristopherBiscardi/gatsby-mdx/blob/05f8eacf5b8f81e5ea521b672673c7762ae2786a/examples/kitchen-sink/gatsby-ssr.js#L8) function at some point but otherwise looks great. Definitely better than it was before this PR :)

**Comment by macklinu** [comment]
Cool, leaving a comment here so I follow up later to break out some issues related to this PR.

**Pull request merged**